File: balanced-binary-tree/balanced-binary-tree.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    int height(TreeNode* ptr){
        if(ptr==NULL){
            return 0;
        }
        return 1+ max(height(ptr->left),height(ptr->right));
    }
    bool isBalanced(TreeNode* root) {
        if(root==NULL){
            return true;
        }
        int LeftTree = height(root->left);
        int RightTree = height(root->right);
        if(abs(LeftTree-RightTree)>1){
            return false;
        }
        
        return isBalanced(root->left) && isBalanced(root->right);
    }
};

File: balanced-binary-tree/balanced-binary-tree.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    private int height(TreeNode head){
        if(head == null){
            return 0;
        }
        return 1 + Math.max(height(head.left), height(head.right));
    }
    public boolean isBalanced(TreeNode root) {
        if(root == null ){
            return true;
        }
        int leftTree = height(root.left);
        int rightTree = height(root.right);
        if (Math.abs(leftTree - rightTree) > 1) {
            return false;
        }
        return isBalanced(root.left) && isBalanced(root.right);
    }
}